src/app/locator/qgsactivelayerfeatureslocatorfilter.cpp

```diff
@@ -271,8 +287,29 @@ void QgsActiveLayerFeaturesLocatorFilter::triggerResult( const QgsLocatorResult
       if ( layer )
       {
         QgsFeatureId fid = data.value( QStringLiteral( "feature_id" ) ).value<QgsFeatureId>();
-        QgisApp::instance()->mapCanvas()->zoomToFeatureIds( layer, QgsFeatureIds() << fid );
-        QgisApp::instance()->mapCanvas()->flashFeatureIds( layer, QgsFeatureIds() << fid );
+        if ( actionId == OpenForm || !data.value( QStringLiteral( "layer_is_spatial" ), true ).toBool() )
```

Comment:
The "all layers" filter doesn't have this same logic -- it just tries to flash/zoom the features for non-spatial layers too. Is that a bug?

Reply:
I'd say so yes. I can fix this one too.

Reply:
done